satpy/readers/viirs_l1b.py

```diff
@@ -235,6 +235,11 @@ def get_dataset(self, dataset_id, ds_info):
         if factors[0] != 1 or factors[1] != 0:
             data *= factors[0]
             data += factors[1]
+
+        # update scale_factor and scale_offset
+        data.attrs["scale_factor"] = scale_factor
+        data.attrs["add_offset"] = scale_offset
```

Comment:
Correct me if I'm wrong but aren't these `None` for BT data (`units == "K"`)?

Reply:
These values are assigned as 1 and 0 here: https://github.com/pytroll/satpy/blob/7a0dc1ad024dc96f30f533413378e61e04cf6abb/satpy/readers/viirs_l1b.py#L78

Reply:
Yes, but those are returned by that method as `factors = self.adjust_scaling_factors` so you would need to use `data.attrs["scale_factor"] = factors[0]`, right? And similarly `factors[1]` for the offset.

Reply:
Now for you to write a test that checks these values :wink: